toolkits/x/arcade_x/tools/users.py

```diff
@@ -1,64 +1,79 @@
 from typing import Annotated
 
 import httpx
-
 from arcade.sdk import ToolContext, tool
 from arcade.sdk.auth import X
-from arcade.sdk.errors import ToolExecutionError
-from arcade_x.tools.utils import expand_urls_in_user_description, expand_urls_in_user_url
+from arcade.sdk.errors import RetryableToolError, ToolExecutionError
+
+from arcade_x.tools.utils import (
+    expand_urls_in_user_description,
+    expand_urls_in_user_url,
+    get_headers_with_token,
+)
+
+# Users Lookup Tools. See developer docs for additional available query parameters:
+# https://developer.x.com/en/docs/x-api/users/lookup/api-reference
 
 
-# Users Lookup Tools. See developer docs for additional available query parameters: https://developer.x.com/en/docs/x-api/users/lookup/api-reference
 @tool(requires_auth=X(scopes=["users.read", "tweet.read"]))
 async def lookup_single_user_by_username(
     context: ToolContext,
     username: Annotated[str, "The username of the X (Twitter) user to look up"],
 ) -> Annotated[dict, "User information including id, name, username, and description"]:
     """Look up a user on X (Twitter) by their username."""
 
-    headers = {
-        "Authorization": f"Bearer {context.authorization.token}",
-    }
-    url = f"https://api.x.com/2/users/by/username/{username}?user.fields=created_at,description,id,location,most_recent_tweet_id,name,pinned_tweet_id,profile_image_url,protected,public_metrics,url,username,verified,verified_type,withheld,entities"
+    headers = get_headers_with_token(context)
+
+    user_fields = ",".join([
+        "created_at",
+        "description",
+        "id",
+        "location",
+        "most_recent_tweet_id",
+        "name",
+        "pinned_tweet_id",
+        "profile_image_url",
+        "protected",
+        "public_metrics",
+        "url",
+        "username",
+        "verified",
+        "verified_type",
+        "withheld",
+        "entities",
+    ])
+    url = f"https://api.x.com/2/users/by/username/{username}?user.fields={user_fields}"
 
-    async with httpx.AsyncClient() as client:
-        response = await client.get(url, headers=headers, timeout=10)
+    try:
```

Comment:
Since the update to error handling for tools I've been avoiding try/catch wrapped around API calls.